libs/dex/src/test/java/io/crate/data/AsyncFlatMapBatchIteratorTest.java

```diff
@@ -123,4 +125,70 @@ public void testFlatMapBatchIteratorFullFillsContracts() throws Exception {
             )
         );
     }
+
+    @Test
+    public void test_repeats_source_after_intermediate_move_to_start() throws Exception {
+        var source = TestingBatchIterators.range(1, 4);
+        AsyncFlatMapper<Row, Row> duplicateRow = (row, isLast) -> {
```

Comment:
why do we need to duplicate the Rows?

Reply:
There is no need to do that, it's just a dummy mapper function. It could also return a single row, or three. Or change the values ... it's the first thing that came to my mind

Reply:
thx!